apollo-router/tests/integration/postgres.rs

```diff
@@ -136,40 +138,68 @@ async fn entity_cache_basic() -> Result<(), BoxError> {
         .next_response()
         .await
         .unwrap();
-    insta::assert_json_snapshot!(response);
+    insta::assert_json_snapshot!(response, {
+        ".extensions.apolloCacheDebugging.data[].cacheControl.created" => 0
+    });
 
     let cache_key = format!(
         "{namespace}-version:1.0:subgraph:products:type:Query:hash:6422a4ef561035dd94b357026091b72dca07429196aed0342e9e32cc1d48a13f:data:d9d84a3c7ffc27b0190a671212f3740e5b8478e84e23825830e97822e25cf05c"
     );
-    let s: Record = sqlx::query_as!(
-        Record,
-        "SELECT data FROM cache WHERE cache_key = $1",
-        cache_key
-    )
-    .fetch_one(&mut conn)
-    .await
-    .unwrap();
-    let v: Value = serde_json::from_str(&s.data).unwrap();
-    insta::assert_json_snapshot!(v);
+
+    let mut record = None;
+    // Because insert is async
+    for _ in 0..10 {
+        if let Ok(resp) = sqlx::query_as!(
+            Record,
+            "SELECT data FROM cache WHERE cache_key = $1",
+            cache_key
+        )
+        .fetch_one(&mut conn)
+        .await
+        {
+            record = Some(resp);
+            break;
+        }
+    }
+    match record {
+        Some(s) => {
+            let v: Value = serde_json::from_str(&s.data).unwrap();
+            insta::assert_json_snapshot!(v);
+        }
+        None => panic!("cannot get cache key {cache_key}"),
+    }
 
     let cache_key = format!(
         "{namespace}-version:1.0:subgraph:reviews:type:Product:entity:72bafad9ffe61307806863b13856470e429e0cf332c99e5b735224fb0b1436f7:representation::hash:3cede4e233486ac841993dd8fc0662ef375351481eeffa8e989008901300a693:data:d9d84a3c7ffc27b0190a671212f3740e5b8478e84e23825830e97822e25cf05c"
     );
-    let s: Record = sqlx::query_as!(
-        Record,
-        "SELECT data FROM cache WHERE cache_key = $1",
-        cache_key
-    )
-    .fetch_one(&mut conn)
-    .await
-    .unwrap();
-    let v: Value = serde_json::from_str(&s.data).unwrap();
-    insta::assert_json_snapshot!(v);
+    let mut record = None;
+    // Because insert is async
+    for _ in 0..10 {
```

Comment:
The ~same 20 lines block of code is used multiple times, it could be a helper function

Reply:
Fixed